compiler/rustc_middle/src/mir/interpret/allocation/provenance_map.rs

```diff
@@ -195,6 +195,28 @@ impl<Prov: Provenance> ProvenanceMap<Prov> {
 
         Ok(())
     }
+
+    pub fn write_wildcards(&mut self, alloc_size: usize, cx: &impl HasDataLayout) {
+        // We can only write wildcards in Miri.
+        assert!(
+            Prov::OFFSET_IS_ADDR,
+            "writing wildcard provenance is not supported when `OFFSET_IS_ADDR` is false"
+        );
+        let wildcard = Prov::WILDCARD.unwrap();
+        let ptr_size = cx.data_layout().pointer_size;
+
+        // Write wildcards in intervals of pointer size.
+        let end = Size::from_bytes(alloc_size / ptr_size.bytes_usize());
+        for offset in Size::ZERO..end {
+            self.ptrs.insert(offset, wildcard);
```

Comment:
This isn't right. You are adding overlapping pointers here! `offset` is in unit of bytes, not 8-bytes.

You could make your life easier by only using per-byte provenance, and leaving `ptrs` empty.

Reply:
Yes - not sure what I was thinking, I think I even had a version that correctly stepped by `pointer_size` - I tried to fix this now by clearing `self.ptrs` and only writing to `self.bytes`.

Reply:
Yeah that is the much easier approach.